accounts/service.go

```diff
@@ -83,16 +85,28 @@ func NewService(dir string, errChan chan<- error) (*InterceptorService, error) {
 		pendingPayments:  make(map[lntypes.Hash]*trackedPayment),
 		mainErrChan:      errChan,
 		quit:             make(chan struct{}),
+		isEnabled:        false,
 	}, nil
 }
 
 // Start starts the account service and its interceptor capability.
 func (s *InterceptorService) Start(lightningClient lndclient.LightningClient,
-	routerClient lndclient.RouterClient, params *chaincfg.Params) error {
+	routerClient lndclient.RouterClient,
+	params *chaincfg.Params) (returnErr error) {
 
 	s.routerClient = routerClient
 	s.checkers = NewAccountChecker(s, params)
 
+	s.isEnabled = true
+
+	defer func() {
+		if returnErr != nil {
+			s.Lock()
+			s.disable()
+			s.Unlock()
```

Comment:
I suggest letting `disable` handle the locking and unlocking of the mutex itself. And then you can have another `disableUnsafe` method which does not grab the mutex so that you can use that one in places where the mutex is already acquired. 

```
func (s *InterceptorService) disable() {
	s.Lock()
	defer s.Unlock()
	
	s.disableUnsafe()
}

func (s *InterceptorService) disableUnsafe() {
	s.isEnabled = false
}
```

Reply:
Also, it's a bit of a weird pattern with the `defer` and `returnErr` (pretty cool idea in general though!).
I would make it a bit more explicit by just replacing any `return fmt.Errorf()` with `return disableAndErrorf(...)`:
```
disableAndErrorf := func (format string, a ...any) error {
    s.disable()
    return fmt.Errorf(format, a...)
}
```